esx_service/utils/auth.py

```diff
@@ -1,4 +1,199 @@
 import random
+import logging
+import auth_data
+import os
 
+auth_mgr = None
+def connect_auth_db():
+    global auth_mgr
+    if not auth_mgr:
+        db_path = os.path.join(auth_data.get_auth_db_path())
+        auth_mgr = auth_data.AuthorizationDataManager(db_path)
+        auth_mgr.connect()
+       
+def find_tenant_by_vm(vm_uuid):
+    """
+        Find tenant which owns this VM by querying the auth DB
+    """
+    global auth_mgr
+    cur = auth_mgr.conn.execute(
+                "select * from vms where vm_id=?",
+                (vm_uuid,)
+    )
+    tenant_uuid = None
+    tenant_name = None
+    result = cur.fetchone()
+    if result:
+        tenant_uuid = result[2]
```

Comment:
Should use the `Row` accessor and retrieve the column values by name instead of index. It is much clearer that way.
https://pysqlite.readthedocs.io/en/latest/sqlite3.html#sqlite3.Row
https://pysqlite.readthedocs.io/en/latest/sqlite3.html#sqlite3.Connection.row_factory